news-TS/src/types/index.ts

```diff
@@ -0,0 +1,62 @@
+export interface IArticle {
+    source: ISource;
+    author: string;
+    title: string;
+    description: string;
+    url: string;
+    urlToImage: string;
+    publishedAt: string;
+    content: string;
+}
+
+export interface ISource {
+    id: string;
+    name: string;
+    description?: string;
+    url?: string;
+    category?: string;
+    language?: string;
+    country?: string;
+}
+
+export interface IApiResponse {
+    status: string;
+    sources?: ISource[];
+    articles?: IArticle[];
+}
+
+export interface IDraw<T> {
+    draw: (value: T[]) => void;
+}
+
+export type apiRequest = {
+    endpoint: endpoints;
+    options?: apiOptions;
+};
+
+export type apiOptions = {
+    apiKey?: string;
+    sources?: string;
+    language?: string;
+    country?: string;
+};
+
+export type endpoints = 'sources' | 'everything' | 'top-headlines';
+
+export type drawContentFunc = (data: IApiResponse) => void;
+
+export enum httpErr {
```

Comment:
In programming there is an unspoken agreement to use uppercase for constants and enums:
```
export enum HTTP_ERR {
  UNATHORIZED = 401,
  NOT_FOUND = 404,
}
```